### PR TITLE
enhancement: aggregate build commands into Makefile

### PR DESCRIPTION
**Pull request by ks6088ts**
# Description
This PR includes following changes:
- codes scattered around build scripts and GitHub Actions are aggregated into Makefile
- fix lint issues by adding comments for exported functions
- run lint and collect code complexity metrics just as goreportcard does

# Checklist
- [x] run ci test locally by `make ci-build-artifacts`
- [x] `./scripts/build.sh 1.2 && ./test/test.sh 1.2.3`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+.DEFAULT_GOAL := help
+
+install-dev: ## Install dev dependencies
+	@echo 'Installing dependencies for development'
+	go get -u \
+		golang.org/x/lint/golint \
+		github.com/fzipp/gocyclo/cmd/gocyclo
+.PHONY:install-dev
+
+install: ## Install dependencies
```
Comment (bearmini) [inline Makefile:25]: How about using `install-deps` instead? Some people might think `install` installs `soracom-cli` itself to somewhere.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,85 @@
+VERSION ?= 0.0.0
+GOOS ?= darwin
+GOARCH ?= amd64
```
Comment (bearmini) [inline Makefile:3]: Can we have these default values for the environment running `make`, by using `uname` or something? (I'm using Ubuntu for my daily work, so I don't want to specify GOOS=linux everytime :P)

Comment (ks6088ts) [reply Makefile]: Thanks for the comment. I use `go env` to retrieve these variables, instead of hardcoding.  17ebc5a

**Review by bearmini — APPROVED** [review]
Thank you for the great work!

**Pull request merged**